sdk/core/azure-core/tests/test_streaming.py

```diff
@@ -92,6 +92,22 @@ def test_decompress_compressed_no_header(http_request):
     except UnicodeDecodeError:
         pass
 
+@pytest.mark.parametrize("http_request", HTTP_REQUESTS)
+def test_compress_compressed_no_header_offline(port, http_request):
+    # expect compressed text
+    client = PipelineClient("")
+    try:
+        request = http_request(method="GET", url="http://localhost:{}/streams/compressed_no_header".format(port))
+        pipeline_response = client._pipeline.run(request, stream=True)
+        response = pipeline_response.http_response
+        data = response.stream_download(client._pipeline, decompress=False)
+        content = b"".join(list(data))
+        with pytest.raises(UnicodeDecodeError):
+            decoded = content.decode('utf-8')
+    except HttpResponseError as e:
+        print(e.response.text())
+
+@pytest.mark.live_test_only
```

Comment:
the async test is https://github.com/Azure/azure-sdk-for-python/blob/33c845aa7674c26bf851313daeae23e2f3f5459c/sdk/core/azure-core/tests/async_tests/test_streaming_async.py#L94